src/test/java/com/amazon/ion/impl/IonRawTextWriterTest_1_1.kt

```diff
@@ -0,0 +1,1022 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.impl
+
+import com.amazon.ion.*
+import com.amazon.ion.system.*
+import java.math.BigDecimal
+import java.math.BigInteger
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+
+class IonRawTextWriterTest_1_1 {
+
+    private inline fun writeAsString(
+        builderConfigurator: IonTextWriterBuilder.() -> Unit = { /* noop */ },
+        autoClose: Boolean = true,
+        block: IonRawTextWriter_1_1.() -> Unit,
+    ): String {
+        val out = StringBuilder()
+        val b = IonTextWriterBuilder.standard()
+            .apply(builderConfigurator)
+            // Always use LF because the tests' expected data uses LF.
+            .withNewLineType(IonTextWriterBuilder.NewLineType.LF)
+
+        val rawWriter = IonRawTextWriter_1_1(
+            options = b as _Private_IonTextWriterBuilder,
+            output = _Private_IonTextAppender.forAppendable(out)
+        )
+        block.invoke(rawWriter)
+        if (autoClose) rawWriter.close()
+        return out.toString()
+    }
+
+    private inline fun assertWriterOutputEquals(
+        text: String,
+        builderConfigurator: IonTextWriterBuilder.() -> Unit = { /* noop */ },
+        autoClose: Boolean = true,
+        block: IonRawTextWriter_1_1.() -> Unit,
+    ) {
+        assertEquals(text, writeAsString(builderConfigurator, autoClose, block))
+    }
+
+    private inline fun assertWriterThrows(block: IonRawTextWriter_1_1.() -> Unit) {
```

Comment:
Will the test succeed if *any* line in the block fails with the expected exception type? Should this accept two blocks: one for any setup code, and one for the line that is expected to fail?

Reply:
> Will the test succeed if any line in the block fails with the expected exception type?

Yes.

> Should this accept two blocks: one for any setup code, and one for the line that is expected to fail?

It might be a little uglier code, but I can make that change.